lib/active_model/serializer/attribute.rb

```diff
@@ -0,0 +1,13 @@
+module ActiveModel
+  class Serializer
+    Attribute = Struct.new(:name, :key, :block) do
+      def value(serializer)
+        if block
+          serializer.instance_eval(&block)
+        else
+          serializer.read_attribute_for_serialization(name)
+        end
+      end
```

Comment:
I feel better about this l, since it appears to be a pattern we're using all over. However, I still don't have a good sense if we'll want to keep this style, and would like to mark the object as part of the internal api.